src/components/admin/form-components/MaxLenTextField.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { capFirstLetter } from 'lib/utilities';
+import TextField from 'material-ui/TextField';
+
+export default class MaxLenTextField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  onChange(event) {
+    this.props.onUpdate(event.target.value.substr(0, this.props.maxLen));
+  }
+
+  render() {
+    return (
+      <TextField
+        name={this.props.name}
+        floatingLabelText={
+          `${capFirstLetter(this.props.name)} (${this.props.value.length}` +
+                           ` of ${this.props.maxLen} characters)`
```

Comment:
I find this indentation weird

Reply:
I usually leave indentation to `eslint --fix`, but yeah this is weird - TODO